src/RxChannel.ts

```diff
@@ -241,10 +241,8 @@ class RxChannel {
 
       return () => {
         closeSub.unsubscribe();
-        try {
-          this.cancel(tag);
-        } catch (e) {
-        } // This prevents a race condition
+        this.cancel(tag)
+          .onErrorResumeNext(Observable.of(false));
```

Comment:
@SkippyZA Not very sure on this change, TBH. Your `try/catch` is synchronous while this isn't, so I'm not exactly confident it'll swallow exceptions. I couldn't reproduce the "race condition" you mention on the comment.

An alternative is just to `return false` on the `catch` block.